app/components/Login/Login.jsx

```diff
@@ -47,6 +47,13 @@ export default class Login extends React.Component {
             this.setState({
                 openSettings: true
             });
+        } else {
+            // If a token was supplied in the window.suppliedAuthToken variable, then simulate a login
+            if ( window.suppliedAuthToken ) {
+                this.setState({loginMethodType: 'TOKEN', authToken: window.suppliedAuthToken}, function () {
+                    this.validateToken({keyCode: 13, tokenSupplied: true});
```

Comment:
this logic looks like it belongs in the constructor, can we move it there instead.